### DISCUSSION TITLE
Add a function to read streamline from the result generated by the command "probtrackx2" in FMRIB's Diffusion Toolbox

### DISCUSSION THREAD

**Discussion by forwho** [dipy/dipy#2395] [Tractography / Tractometry] [answered]
Adding the option "--savepaths" when using the command "probtrackx2" in FMRIB's Diffusion Toolbox, the program will save the streamline coordinates in a text file. Could developers of DIPY add a function to read the text file and save the streamlines to a Tractogram object.

**Comment by skoudoro** [comment, marked as answer]
Hi @forwho,

Can you provide an example of this text file?

I am not sure we will add this function and I am surprised that they do not provide a command to convert their text file to a standard tractography file like tck/trk.

I recommend you to ask the question on [Neurostars](https://neurostars.org/). I supposed someone already faces this problem and I am pretty sure a converter already exists.

Please, keep us updated and we will look at your text file

**Reply by forwho** [reply]
Hi @skoudoro ,

Please see the attached for an example of the text file. I guess the number after # is the number of steps used for the track creation. And the following lines are voxel coordinates.

The FSL discourages to save streamlines. So they do not provide a command to covert their text file to tck/trk.

[saved_paths1.txt](https://github.com/dipy/dipy/files/6505066/saved_paths1.txt)

**Reply by forwho** [reply]
Hi @skoudoro 

I have developed the function that can convert the text file to the trk file. 

Thanks anyway. If you want to add this function, I can push the code to the DIPY repository.

**Reply by skoudoro** [reply]
> I have developed the function that can convert the text file to the trk file.

That's great! thank you for letting us know. And yes, if you could push the code to the DIPY repository or just provides the link of your code and someone will do it, it would be nice.

Thank you @forwho!

**Comment by skoudoro** [comment]
Also, I am moving this question on [our forum](https://github.com/dipy/dipy/discussions)
